File: frontend/lib/models/role.dart
enum Role {
  user,
  admin;

  static Role? fromString(String? value) {
    if (value == null) return null;
    try {
      return Role.values.firstWhere(
        (role) => role.name.toLowerCase() == value.toLowerCase(),
      );
    } catch (e) {
      return null;
    }
  }

  String toDisplayString() {
    switch (this) {
      case Role.user:
        return 'User';
      case Role.admin:
        return 'Admin';
    }
  }
}

File: frontend/src/shared/types/role.ts
export type Role = 'user' | 'admin';

export const roleFromString = (value?: string | null): Role | null => {
  if (!value) return null;
  const lower = value.toLowerCase();
  if (lower === 'user') return 'user';
  if (lower === 'admin') return 'admin';
  return null;
};

export const roleDisplay = (role?: Role | null): string => {
  if (role === 'admin') return 'Admin';
  return 'User';
};
